src/providers/python_controller.js

```diff
@@ -1,31 +1,26 @@
-import {execSync} from "node:child_process";
 import fs from "node:fs";
 import path from 'node:path';
 import os, {EOL} from "os";
-import {environmentVariableIsPopulated,getCustom, handleSpacesInPath} from "../tools.js";
-
+import {environmentVariableIsPopulated,getCustom, invokeCommand} from "../tools.js";
 
 function getPipFreezeOutput() {
-	return environmentVariableIsPopulated("EXHORT_PIP_FREEZE")  ? new Buffer(process.env["EXHORT_PIP_FREEZE"],'base64').toString('ascii') : execSync(`${handleSpacesInPath(this.pathToPipBin)} freeze --all`, err => {
-		if (err) {
-			throw new Error('fail invoking pip freeze to fetch all installed dependencies in environment --> ' + err.message)
-		}
-	}).toString();
+	try {
+		return environmentVariableIsPopulated("EXHORT_PIP_FREEZE")  ? new Buffer(process.env["EXHORT_PIP_FREEZE"], 'base64').toString('ascii') : invokeCommand(this.pathToPipBin, ['freeze', '--all']).toString();
+	} catch (error) {
+		throw new Error('fail invoking pip freeze to fetch all installed dependencies in environment --> ' + error.message)
```

Comment:
in other places we started using `{cause: error}` 
```suggestion
		throw new Error('Failed to invoke pip freeze to fetch all installed dependencies in environment', { cause: error });
```